classes/task/delete_orphaned_object_metadata.php

```diff
@@ -52,10 +52,28 @@ public function execute() {
             'location' => OBJECT_LOCATION_ORPHANED,
             'ageforremoval' => time() - $ageforremoval
         ];
-        $count = $DB->count_records_select('tool_objectfs_objects', $wheresql, $params);
-        if (!empty($count)) {
-            mtrace("Deleting $count records with orphaned metadata (orphaned tool_objectfs_objects)");
-            $DB->delete_records_select('tool_objectfs_objects', $wheresql, $params);
+
+        if (!empty($this->config->deleteexternal) && $this->config->deleteexternal == TOOL_OBJECTFS_DELETE_EXTERNAL_TRASH) {
+            // We need to delete the external files as well as the orphaned data.
+            $filesystem = new $this->config->filesystem();
+            $objects = $DB->get_recordset_select('tool_objectfs_objects', $wheresql, $params);
```

Comment:
I'm tempted to sanity check this by using some sql that will only return tool_objectfs_objects where the contenthash doesn't exist in the files table - @brendanheywood that make sense to you?